src/SablierV2MerkleStreamerFactory.sol

```diff
@@ -36,33 +31,53 @@ contract SablierV2MerkleStreamerFactory is ISablierV2MerkleStreamerFactory {
         // Hash the parameters to generate a salt.
         bytes32 salt = keccak256(
             abi.encodePacked(
-                initialAdmin,
+                params.initialAdmin,
+                params.asset,
+                bytes32(abi.encodePacked(params.name)),
+                params.merkleRoot,
+                params.expiration,
+                params.cancelable,
+                params.transferable,
                 lockupLinear,
-                asset,
-                merkleRoot,
-                expiration,
-                abi.encode(streamDurations),
-                cancelable,
-                transferable
+                abi.encode(streamDurations)
             )
         );
 
         // Deploy the Merkle streamer with CREATE2.
-        merkleStreamerLL = new SablierV2MerkleStreamerLL{ salt: salt }(
-            initialAdmin, lockupLinear, asset, merkleRoot, expiration, streamDurations, cancelable, transferable
-        );
+        merkleStreamerLL = new SablierV2MerkleStreamerLL{ salt: salt }(params, lockupLinear, streamDurations);
+
+        // Using a different function to emit the event to avoid stack too deep error.
+        _emitLLEvent(merkleStreamerLL, params, lockupLinear, streamDurations, ipfsCID, aggregateAmount, recipientsCount);
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                          INTERNAL NON-CONSTANT FUNCTIONS
+    //////////////////////////////////////////////////////////////////////////*/
 
+    /// @dev Helper function to emit the {CreateMerkleStreamerLL} event.
+    function _emitLLEvent(
```

Comment:
Why are we not simply emitting the `params` struct in the event instead of doing this?

It should be pretty easy to parse that struct in the subgraph (since the subgraph has to be able to parse structs anyway).

Cc @razgraf can lend us a hand here?

Reply:
I guess we could, are there any variables we're not emitting (for gas purposes) in the iterative way that we would by replacing it with the params structs?

Reply:
imo, here is it not about gas, but about DX

would it affect the dev work?

Reply:
We're adding a new parameter anyway so the event has to be duplicated regardless. Of course, changing from a list into a single struct parameter will make union types e.g. `event: Event1 | Event2` harder to process but it's a minor point in the grand scheme.

Reply:
ok, got it

then, we will emit the struct